src/internal/storage/chunk/storage.go

```diff
@@ -67,6 +67,11 @@ func (s *Storage) NewDataReader(ctx context.Context, dataRef *DataRef) io.Reader
 	return newDataReader(ctx, client, s.memCache, s.deduper, dataRef, 0)
 }
 
+func (s *Storage) PrefetchData(ctx context.Context, dataRef *DataRef) error {
```

Comment:
Sidenote, if you changed `NewDataReader()` to:

``` go
func (s *Storage) NewDataReader(ctx context.Context, dataRef *DataRef) *DataReader {
	client := NewClient(s.store, s.db, s.tracker, nil)
	return newDataReader(ctx, client, s.memCache, s.deduper, dataRef, 0)
}
```

You could change `PrefetchData()` to: 
```go
func (s *Storage) PrefetchData(ctx context.Context, dataRef *DataRef) error {
	return s.NewDataReader(ctx, dataRef).fetchData()
}
```

It seems like `DataReader` is already exported and implements the io.Reader interface, so this shouldn't break anything. Structuring it this may might help make `PrefetchData` resilient to changes made to `NewClient`'s signature.